### ISSUE TITLE
Backing Property rule not work.

### ISSUE THREAD

**Issue by mikanIchinose** [pinterest/ktlint#2077]
## Expected Behavior

```kotlin
class Foo {
    private val mElementList = mutableListOf<Element>()
                                                       
    val elementList: List<Element>
        get() = mElementList
}
```

--> `Backing property name should start with underscore followed by lower camel case`

## Observed Behavior

There are no errors.

## Steps to Reproduce

1. clone pinterest/ktlint repository
2. open `ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/rules/PropertyNamingRuleTest.kt`
3. add test like bellow

```kotlin
@Test
fun `hoge`() {
    val code =
        """
        class Foo {
            private val mElementList = mutableListOf<Element>()
                                                               
            val elementList: List<Element>
                get() = mElementList
        }
        """.trimIndent()
    propertyNamingRuleAssertThat(code).hasNoLintViolations()
}
```

## Your Environment

* Operating System and version: macOS 13.4

**Comment by paul-dingemans** [comment]
Klint has only limited knowledge (only that what is available in the AST and PSI) when parsing the code. This sometimes limits what is possible or not.

The definition of a backing property is a bit vague. According to the [Kotlin Coding conventions](https://kotlinlang.org/docs/coding-conventions.html#names-for-backing-properties):

> If a class has two properties which are conceptually the same but one is part of a public API and another is an implementation detail, use an underscore as the prefix for the name of the private property:

"Conceptually the same" is something that can not be determined by Ktlint. So the only check that is now in place is to verify that whenever a private property is found for which the name starts with `_` and for which a public property exists with same name not starting with `_` then the name should match the pattern for backing properties.

**Comment by mikanIchinose** [comment]
Thanks. I understand!

```kotlin
// Wrong
class Foo {
  private val _elementList = mutableListOf<Element>()
}
// rule: standard:property-naming
// Property name should start with a lowercase letter and use camel case

// Ok
class Foo {
  private val _elementList = mutableListOf<Element>()
  val elementList: List<Element>
      get() = _elementList
}
// There is no error because naming pattern is matched `backing property`
```

**Comment by paul-dingemans** [comment]
Tnx for feedback. I have created a new issue to add this to the [rule documentation](https://pinterest.github.io/ktlint/0.49.1/rules/experimental/#property-naming).
